tests/test-runner/bin/test-runner.py

```diff
@@ -288,6 +289,10 @@ def run(self, dryrun):
             fail('\nRun terminated at user request.')
         finally:
             t.cancel()
+            # Clear any leftover injected badness, just in case.
```

Comment:
I could see us wanting to do other things of this nature. What would you think about allowing the framework to specify a user script (or scripts) to be run when a test is killed? I think it would be good to keep zfs specific logic out of the test runner if possible.

Reply:
I agree. Deciding how to achieve this is the next challenge.